storage/rocksdb/rdb_datadic.cc

```diff
@@ -185,6 +185,18 @@ int Rdb_convert_to_record_key_decoder::skip(
   if (fpi->m_skip_func == &Rdb_key_def::skip_variable_space_pad &&
       !fpi->m_unpack_info_stores_value && !covered_bitmap_format_enabled) {
     unp_reader->read(fpi->m_unpack_info_uses_two_bytes ? 2 : 1);
+  } else if (fpi->m_covered == Rdb_key_def::KEY_MAY_BE_COVERED &&
```

Comment:
I just started wondering if these 2 cases are not "hacks" that should be handled by proper `m_skip_func` function.
`m_skip_func` is used in several other places. Don't we have the same problem of 'partial skip' there?

Reply:
@kamil-holubicki  It would be a bit tricky to re-implement skip function, because it does not take `unp_reader()` as an argument. It looks that other use of `m_skip_function()` are safe because they do not combine pack and unpack reader or they also skip unpacked bytes (`unpack_unknown_varlength()`).